File: tests/fixtures/basic_graphql.php
<?php

$getUserGql = <<<'GQL'
      query GetUser {
        user {
          id
          name
          email
        }
      }
    GQL;

$createUserGql = <<<'GQL'
      mutation CreateUser {
        createUser {
          id
          name
          email
        }
      }
    GQL;

$userUpdatedGql = <<<'GQL'
      subscription OnUserUpdated {
        userUpdated {
          id
          name
          status
        }
      }
    GQL;

$userFieldsGql = <<<'GQL'
      fragment UserFields on User {
        id
        name
        email
      }
    GQL;

$searchUsersGql = <<<'GQL'
      query SearchUsers {
        activeUsers: users(status: ACTIVE) {
          ...UserFields
        }
        inactiveUsers: users(status: INACTIVE) @include(if: true) {
          ...UserFields
        }
      }
    GQL;

$getEntityGql = <<<'GQL'
      query GetEntity {
        entity {
          __typename
          ... on User {
            id
            name
          }
          ... on Organization {
            id
            displayName
          }
        }
      }
    GQL;

File: tests/fixtures/basic_graphql.java
class GraphQLMain {
  void run() {
    String GET_USER_GQL =
        """
        query GetUser($id: ID!) {
          user(id: $id) {
            id
            name
            email
          }
        }
        """;

    String createUserGql =
        """
        mutation CreateUser($input: CreateUserInput!) {
          createUser(input: $input) {
            id
            name
            email
          }
        }
        """;

    String userUpdatedGql =
        """
        subscription OnUserUpdated($userId: ID!) {
          userUpdated(userId: $userId) {
            id
            name
            status
          }
        }
        """;

    String userFieldsGql =
        """
        fragment UserFields on User {
          id
          name
          email
        }
        """;

    String searchUsersGql =
        """
        query SearchUsers($query: String!, $includeInactive: Boolean!) {
          activeUsers: users(query: $query, status: ACTIVE) {
            ...UserFields
          }
          inactiveUsers: users(query: $query, status: INACTIVE) @include(if: $includeInactive) {
            ...UserFields
          }
        }
        """;

    String getEntityGql =
        """
        query GetEntity($id: ID!) {
          entity(id: $id) {
            __typename
            ... on User {
              id
              name
            }
            ... on Organization {
              id
              displayName
            }
          }
        }
        """;

    System.out.println(GET_USER_GQL);
    System.out.println(createUserGql);
    System.out.println(userUpdatedGql);
    System.out.println(userFieldsGql);
    System.out.println(searchUsersGql);
    System.out.println(getEntityGql);
  }
}
